src/core/processing/qgsprocessingcontext.h

```diff
@@ -90,10 +90,10 @@ class CORE_EXPORT QgsProcessingContext
     void setExpressionContext( const QgsExpressionContext &context ) { mExpressionContext = context; }
 
     /**
-     * Returns a reference to the project used for storing temporary layers during
+     * Returns a reference to the layer store used for storing temporary layers during
      * algorithm execution.
      */
-    QgsProject &temporaryLayerStore() { return tempProject; }
+    QgsMapLayerStore &temporaryLayerStore() { return tempLayerStore; }
```

Comment:
It was more of a reflection that there will always be a temporary layer store (but may not always be a project attached!). But I agree, QObject by pointer seems nicer to me too. I'll fix.